### ISSUE TITLE
Allow control from the local WLAN without internet connection

### ISSUE THREAD

**Issue by cajonKA** [NeatoRobotics/neato-sdk-js#7]
I would really like to control my D5 without the Internet, so that it can be used in locations where no Internet is available (yes, such locations still exist)
Are there any plans to extend the sdk such that it allows control from the local WLAN without internet access ?
Something like this must be possible as the Android App DOES connect locally to the bot to configure the Wifi.

**Comment by marcouberti** [comment]
Hi, your request has a lot of sense, but right now we don't have any plans to extend the control of the robot in that direction.

**Comment by Fuzzillogic** [comment]
I got my D5 as exchange for a broken Botvac 85. While I'm grateful for the exchange, I'm less thrilled that without the use of an app, the functionality of the D5 is severely crippled compared to the old 85. I couldn't even update the firmware of the D5, despite having a USB connector.

I value my privacy, so I don't use Android nor iOS. I also value my wallet and the environment, so I don't want to buy products which become useless if their only single service provider goes down. Too often have I seen devices effectively break down because they are tied to their manufacture's defunct online service, forcing the owner to buy a new device, even though the old one is technically perfectly adequate. 
For those reasons I don't buy things like Nest, Alexa, and as it is, Neato's current offerings.

A local API provides an alternative. A back-up. Bonus points for an option to choose another online service provider.

**Comment by uchagani** [comment]
@marcouberti please reconsider this. My next botvac will not be a Neato without local access.

**Comment by reedbn** [comment]
This is a feature that I would like as well. I don't mind if the robot phones home with anonymous stats on occasion, but I also want my robot to still work locally when my ISP drops (which it does from time to time).

**Comment by uchagani** [comment]
@reesdn I was ready to buy another vacuum and decided to spend my money on an ecovacs vacuum. With the bumper and sucks projects I can control my vacuum without needing internet access.

**Comment by reedbn** [comment]
I was seriously considering an Ecovacs Deebot line, but the IMEI requirement for their default app was a pretty big turn off, since I anticipated I'd use the default app of whatever I got for the first little while. Had I come across https://github.com/torbjornaxelsson/bumper earlier, maybe I'd have gone for it anyway...

**Comment by uchagani** [comment]
Once you set up the bot on your wifi you can block all traffic to their
servers. I'm pretty excited about my ecovac and definitely won't be buying
another Neato until they get their heads out of their ass.

**Comment by Fuzzillogic** [comment]
Using a self-hosted server would suit me. As long as the protocol is specified, which more or less is the case already, and there is an option to change the default server address to a own address, _without requiring Android/iPhone_, it might be sufficient.

Still, mDNS + on-device HTTP REST API, with e.g. the serial as key/password would be even better, as that would require only a client.

The current situation is that I can't even stop the thing from broadcasting its presence and leaving it open for hijacking for an hour whenever I use it, because I can't even register the device, because I can't even start the damn app, Neato thought it would be wise to _require_ Google Play Services (which of course I don't have) before doing anything. >:(

**Comment by prchal** [comment]
Local server is for me very important feature.
Are commands from App simply resend to bot and vice versa? Then it could be easy to make such a server.
Where are maps stored? Does the bot has camera and microphone?

**Comment by sadziu82** [comment]
Anything happening here? If you are not willing to let users use their device without local control please just tell it.

**Comment by vpmv** [comment]
My XV broke down recently and since you can't even run a normal vacuum routine with the newer models, I'm not spending my money on Neato anymore.

**Comment by DrDrebon** [comment]
This should be a legal obligation to be able to use connected objects on a personal LAN

**Comment by alanjames1987** [comment]
@DrDrebon I agree that I want local control but I just vote with my money by not buying the product that doesn't have the features I want.

**Comment by thany** [comment]
@marcouberti 
> Hi, your request has a lot of sense, but right now we don't have any plans to extend the control of the robot in that direction.

At that time, of course you didn't have those plans, as you had no idea people wanted this. Now that you can see people want this, I would expect you to do two things:
1. You have asked yourselves why on good mother Earth you've done things wrong, in comparison to before.
2. Made plans, and begun development on allowing to control the device without internet.

So how is that going? What is the status?

And no "we don't have plans" answer, because then you'd properly embarass yourselves ;)

**Comment by cajonKA** [comment]
I think none of the previous contributors still work for Neato, so the chance to get a Local Remote Control for the D5 seem to be pretty small. :-(
We can just consider this for our next buy :-(

**Comment by imperador** [comment]
Now that the D3-7 line is being replaced by the D8-10, this should be a mandatory thing to implement, because not even the new app will have support for the D3-7 devices.

Please, Neato Robotics, allow local control over the old devices or this will certainly become a major problem for your company's PR (as it is already embarrassing you guys by looking at the 2 stars your new app has on both Android and iOS because of the lack of support).

**Comment by newincpp** [comment]
Barely any activities since 2017 (ans not even a sincle commit in the last 2 years) + the sdk is still in beta. It really feel like something that can can go down any minute and without a local mode it's really scary.
Additionally having something able to connecting itself to internet by itself with no real control over it is a security risk I won't take (all my domotics devices connection can't outside of my home network for those reason and many more).
Please do something.

**Comment by avbfr** [comment]
+1 for this feature.
For all the reasons given previously.

**Comment by tantalusblank** [comment]
+1 again for this.
Was pretty ready to buy one of these, but this is a dealbreaker.

**Comment by bugs181** [comment]
I've got two Neato D7's that I'll probably be selling for cheap (or worse, going to the landfill) and replacing with ANY botvac that can flashed with [Valetudo](https://github.com/Hypfer/Valetudo). I know for a fact that Neato will be losing my business in the future unless they MAKE THIS RIGHT by offering a local API. 

As others have pointed out, it's really quite simple. The hacking, hobbyist, and DIY community is SHOWING the Neato company up and proving how things aught to be done. The DIY community created a method of hooking a serial (raspberry pi) device up to the original non-connected botvacs and MADE them wireless WITH local control. Neato didn't like that, so they removed the serial lines and locked it down with a USB port, which is limited. 

SHAME ON YOU, Neato! You just lost a customer that has spent over $3000 on your products.

**Comment by Fuzzillogic** [comment]
[Neato shutting down operations](https://www.techhive.com/article/1803382/neato-robotics-is-shutting-down-robot-vacuums.html). Before you turn off the light, will you please unshackle our vacuums from the cloud? Or are you actually want to serve as a case study in why one shouldn't trust any cloud-operated service ever again?

**Comment by Robbe-B** [comment]
+1 especially now

**Comment by ashfran** [comment]
This behavior of not making products like these accessible locally is what is slowing my investment in devices that solely use cloud services. I've had wink, harmony, dlink and other IoT devices make me bitter. Now neato. I pay for cloud service for arlo, nest, and nabu casa , yet see that my new home assistant devices (like my reolink cameras that are replacing nest)don't come into my house without local and cloud options. Give people that choice.

**Comment by imperador** [comment]
Just to remember: The company behind this is Kobold. They are not shutting down completely, they are just abandoning the users. 

Neato D4-7 are just a rebranded Kobold Vorwerk VR300 with software-locked functionalities, and this is easily proved by the fact that you can install the VR300 firmware on Neato D4-7 with valid certificates being created. The same is valid for the D8-10, as they are just a rebranded Kobold Vorwerk VR7.

If they leave the userbase without adding a possibility of controlling it locally, don't blame "Neato", blame Kobold because that's their name going forward.

**Comment by mattrayner** [comment]
Hi Pawel, the map and history system is working again now, please try and
run a new clean and check for your map :-)

**Comment by pcking99** [comment]
> Just to remember: The company behind this is Kobold. They are not shutting down completely, they are just abandoning their users and pretending they are just "Neato".
> 
> Neato D4-7 are just a rebranded Kobold Vorwerk VR300 with software-locked functionalities, and this is easily proved by the fact that you can install the VR300 firmware on Neato D4-7 with valid certificates being created. The same is valid for the D8-10, as they are just a rebranded Kobold Vorwerk VR7.
> 
> If they leave the userbase without adding a possibility of controlling it locally, don't blame "Neato", blame Kobold because that's their name going forward.

Have you tested flashing with the updates from Vorwerk's website? If thats how to keep our robots alive then we should maybe flash when the server stops working.

**Comment by orcinus** [comment]
It's dead again.
Not making control fully local was a ridiculously bad choice.

**Comment by bugs181** [comment]
It's been some years since I've posted in this thread. 

Although not a solution, more of a "replacement" moving forward - I urge everyone to check out Valetudo's supported botvac list. My Dreame Z10 Pro has been chugging along, completely local for a couple years now! 

I have it automated to clean specific rooms using Home Assistant triggers. It even automatically pauses cleaning when a call comes in and resumes when I hang up. Super cool stuff.

Here's a link to all of the available Botvac's that Valetudo works on.
https://valetudo.cloud/pages/general/supported-robots.html
